pkg/tuned/controller.go

```diff
@@ -450,6 +439,26 @@ func providerExtract(provider string) error {
 	return nil
 }
 
+// Read the Cloud Provider name from openshiftTunedProvider file and
+// extract/write 'provider' only if the file does not exist or it does not
+// match 'provider'.  Returns indication whether the 'provider' changed and
+// an error if any.
+func providerSync(provider string) (bool, error) {
+	providerCurrent, err := os.ReadFile(openshiftTunedProvider)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return len(provider) > 0, providerExtract(provider)
+		}
+		return false, err
+	}
+
+	if provider == string(providerCurrent) {
+		return false, nil
+	}
+
+	return true, providerExtract(provider)
+}
+
 func TunedRecommendFileWrite(profileName string) error {
```

Comment:
I think this function can be simplified quite a lot but this seems to be better done in a followup PR. No changes needed for this PR